src/audio.c

```diff
@@ -185,8 +194,10 @@ static void audio_callback()
  *            The frequency in Hz to play back samples at
  * @param[in] numbuffers
  *            The number of buffers to allocate internally
+ * @param[in] fill_buffer_callback
+ *            A function to be called when more sample data is needed
  */
-void audio_init(const int frequency, int numbuffers)
+void audio_init(const int frequency, int numbuffers, audio_fill_buffer_callback fill_buffer_callback)
```

Comment:
Ooh nice, I like!

Possibly setting the fill buffer in another function instead of the init would be backwards compatible and also less confusing for somebody trying to use, so I like that idea.